python/GafferUI/CompoundEditor.py

```diff
@@ -585,6 +656,33 @@ def __pinningButtonContextMenu( self, button ) :
 
 		return True
 
+	def __addStandardItems( self, editor, m ) :
+
+		# @see CompoundEditor.nodeSetMenuSignal for details on the structure
+		# of this menu.
+
+		drivingEditor, _ = editor.getNodeSetDriver()
+		isPinned  = self.__editorIsPinned( editor )
+
+		isFollowingNodeSelection = drivingEditor is None and not isPinned
+		m.append( "/Following Selected Nodes", {
+			"checkBox" : isFollowingNodeSelection,
+			"command" : None if isFollowingNodeSelection else Gaffer.WeakMethod( self.__followScriptSelection )
+		} )
+
+		m.append( "/Pinned", {
+			"checkBox" : isPinned,
+			"command" : None if isPinned else Gaffer.WeakMethod( self.__pinToScriptSelection )
+		} )
+
+		# Do not change this!
+		# Signals use this to anchor insertions of actions or state items appropriately.
+		m.append( "/ActionsDivider", { "divider" : True } )
+
+		m.append( "/Re-pin Current Node Selection" if isPinned else "Pin Current Node Selection", {
```

Comment:
How about:

`m.append( "/Pin to Current Node Selection", {`

I find the "re" prefix more confusing than anything. Am I pinning to the _same_ node as before? I don't see a reason to include the prefix. In every circumstance, I'm just pinning to whatever's currently selected. The "✓ Pinned" entry and icon should be enough to tell me that I'm already pinned to a node, instead of linked/following.

Reply:
Fixed in https://github.com/GafferHQ/gaffer/commit/aca4e28c8ebb8804581443491d1e9f96db3798b5